src/com/itacademy/eshop/simulations/CustomerSimulation.java

```diff
@@ -29,21 +35,55 @@ private void addProductsToShoppingCart() {
          * User browses the shop and adds 4 products to the shopping cart.
          * One of them should be a book and other should be a shirt.
          */
-        throw new UnsupportedOperationException("addProductsToShoppingCart() is not implemented yet");
+        Scanner fourProductsShoppingScanner = new Scanner(System.in);
+        System.out.println("Please add three new products : first a book ");
```

Comment:
You could have a separate method which read data for one instance of an object and use that method as many times as you need to have instances of an object. This way you will have less code which does the same thing.

Reply:
For example:
`private Product createProduct(Category category) {
Scanner scanner = new Scanner(System.in);
String productName = scanner.nextLine();
int productPrice = Integer.parseInt(scanner.nextLine());
scanner.close();
return new Product(productName, productPrice, category);
}`
This way you can just specify the category as method param and create new instances of Products